### PR TITLE
AAE-29269 Conditional dropdown does not populate after selecting value in a parent dropdown

### PR DESCRIPTION
**Pull request by BSekula**
**Please check if the PR fulfills these requirements**

> - [ ] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/core/src/lib/form/components/form-field/form-field.component.ts [hunk 1]
```diff
@@ -105,9 +108,15 @@ export class FormFieldComponent implements OnInit, OnDestroy {
     }
 
     private updateReactiveFormControlOnFormRulesEvent(instance: any): void {
-        instance?.formService.formRulesEvent.pipe(takeUntilDestroyed(this.destroyRef)).subscribe(() => {
-            instance?.updateReactiveFormControl();
-            this.triggerFormFieldChanged(instance.field);
+        instance?.formService.formRulesEvent.pipe(takeUntilDestroyed(this.destroyRef)).subscribe((event) => {
+            if (
+                event.field.id !== instance.id &&
+                event.type === 'fieldValueChanged' &&
```
Comment (tomgny) [inline lib/core/src/lib/form/components/form-field/form-field.component.ts:114]: what about other event types, like invalid/click etc? do we need this type check here?
image

Comment (BSekula) [reply lib/core/src/lib/form/components/form-field/form-field.component.ts]: changed, triggering update only when value changed

**Comment by mkrbr** [comment]
Do you already have a Unit Test in place?

        let formRulesEvent: Subject<void>;

....

        formRulesEvent = new Subject<void>();
        ....
        TestBed.configureTestingModule({
            imports: [CoreTestingModule],
            providers: [{ provide: 'formService', useValue: { formRulesEvent } }]
        });

....

        it('should call triggerFormFieldChanged once when updateReactiveFormControlOnFormRulesEvent is called', () => {
        const mockForm = new FormModel();
        const field = new FormFieldModel(mockForm, { type: FormFieldTypes.TEXT });
        component.field = field;

        const instance = {
            field,
            formService: { formRulesEvent },
            updateReactiveFormControl: jasmine.createSpy('updateReactiveFormControl')
        };

        spyOn(component as any, 'triggerFormFieldChanged').and.callThrough();

        (component as any).updateReactiveFormControlOnFormRulesEvent(instance);

        formRulesEvent.next();
        fixture.detectChanges();

        expect((component as any).triggerFormFieldChanged).toHaveBeenCalledTimes(1);
    });

**Review by tomgny — APPROVED** [review]

**Comment by BSekula** [comment]
@mkrbr 

https://github.com/Alfresco/alfresco-ng2-components/pull/10484#issuecomment-2535940939

nop, will update one with calling `updateReactiveFormControl` method

**Pull request merged**